dotCMS/src/main/java/com/dotmarketing/db/DBPropertiesDatasourceStrategy.java

```diff
@@ -0,0 +1,86 @@
+    }
+
+    public static DBPropertiesDatasourceStrategy getInstance(){
+        return SingletonHelper.INSTANCE;
+    }
+
+    /**
+     *
+     * @return True if a <b>db.properties</b> file exists in WEB-INF/classes directory
+     */
+    public boolean existsDBPropertiesFile() {
+        final ClassLoader loader = Thread.currentThread().getContextClassLoader();
+        final URL resourceURL = loader.getResource("db.properties");
+        return resourceURL!=null && new File(resourceURL.getPath()).exists();
+    }
+
+    @Override
+    public DataSource getDatasource() {
+        final ClassLoader loader = Thread.currentThread().getContextClassLoader();
+        loader.getResourceAsStream(DB_PROPERTIES_FILE_NAME);
+
+        final PropertiesConfiguration properties = new PropertiesConfiguration();
+        try {
+            properties.load(loader.getResourceAsStream(DB_PROPERTIES_FILE_NAME));
+
+            final HikariConfig config = new HikariConfig();
+
+            config.setPoolName(Constants.DATABASE_DEFAULT_DATASOURCE);
+            config.setDriverClassName(properties.getString("connection_db_driver"));
+            config.setJdbcUrl(properties.getString("connection_db_base_url"));
+            config.setUsername(properties.getString("connection_db_username"));
+            config.setPassword(properties.getString("connection_db_password"));
+            config.setMaximumPoolSize(properties.getInt("connection_db_max_total", 60));
+            config.setIdleTimeout(properties.getInt("connection_db_max_idle", 10) * 1000);
+            config.setMaxLifetime(properties.getInt("connection_db_max_wait", 60000));
+            config.setConnectionTestQuery(properties.getString("connection_db_validation_query"));
+
+            // This property controls the amount of time that a connection can be out of the pool before a message
+            // is logged indicating a possible connection leak. A value of 0 means leak detection is disabled.
+            // Lowest acceptable value for enabling leak detection is 2000 (2 seconds). Default: 0
+            config.setLeakDetectionThreshold(properties.getInt("connection_db_leak_detection_threshold", 60000));
+
+            config.setTransactionIsolation(properties.getString("connection_db_default_transaction_isolation"));
+
+            properties.clear();
+            return new HikariDataSource(config);
+        } catch (ConfigurationException e) {
+            Logger.error(DBPropertiesDatasourceStrategy.class,
+                    "---------- Error getting dbconnection " + Constants.DATABASE_DEFAULT_DATASOURCE
+                            + " from db.properties file",
+                    e);
+            if(Config.getBooleanProperty("SYSTEM_EXIT_ON_STARTUP_FAILURE", true)){
+                e.printStackTrace();
+                System.exit(1);
+            }
+
+            throw new DotRuntimeException(e.toString());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [New exception is thrown in catch block, original stack trace may be lost](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5026767)